packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -799,9 +799,12 @@ export default class NodeResolver {
     if (env.isBrowser()) pkgKeys.push('browser');
 
     for (let pkgKey of pkgKeys) {
-      let alias = await this.getAlias(filename, pkg, pkg[pkgKey]);
-      if (alias != null) {
-        return alias;
+      let pkgKeyValue = pkg[pkgKey];
+      if (!Array.isArray(pkgKeyValue)) {
+        let alias = await this.getAlias(filename, pkg, pkgKeyValue);
+        if (alias != null) {
+          return alias;
+        }
```

Comment:
@devongovett @mischnic When I correctly changed the PackageTargetDescriptor.source to "FilePath | Array<FilePath" it caused an error here which I resolved with the explicit Array check.

I'm going to commit this and it didn't break any unit tests, but we should verify that this is the correct aliasing behavior when a source array is provided.

It looks like if source is a single FilePath then it's an alias, if source is an Array then none of the values are aliases.

Reply:
I didn't know that `source` could also behave like `alias`. @devongovett ?

You've already removed that alias variant in the types package, which previously was
```js
	source?: FilePath | {[FilePath]: FilePath, ...},
```
 So maybe we could just not process `source` here anymore? (And then the added condition isn't needed anymore)

Reply:
Yeah that may have been from Parcel 1. I think the idea was to map compiled code back to source when you use a symlinked package.

Reply:
So, should we remove it or keep it (though it's untested at the moment)?

Reply:
It is documented on the v1 site: https://parceljs.org/transforms.html#third-party-modules.

Whether we want to keep that (and other Parcel specific resolution features) for v2 is questionable. I'm fine with merging as is and deciding outside this PR.